src/providers/ldap/sdap_utils.c

```diff
@@ -206,6 +206,39 @@ char *sdap_combine_filters(TALLOC_CTX *mem_ctx,
     return sdap_combine_filters_ex(mem_ctx, '&', base_filter, extra_filter);
 }
 
+/* we check the principal and if it contains our realm, then we drop it
+ * for the comparison with sAMAccountName */
+
+char *principal_string_to_sAMAccountName(TALLOC_CTX *mem_ctx,
+                                         const char *attr_name,
+                                         const char *princ,
+                                         struct dp_option *sdap_basic_opts)
+{
+    const char *realm;
+    char *p;
+
+    if (attr_name == NULL || princ == NULL || sdap_basic_opts == NULL) {
+        return NULL;
+    }
+
+    realm = dp_opt_get_cstring(sdap_basic_opts, SDAP_KRB5_REALM);
```

Comment:
Hi,
same here, `SDAP_KRB5_REALM` would be wrong for trusted domain. It works for `get_enterprise_principal_string_filte()` because with enterprise principals it is expected that the realm is from the local domain and the request is send to the local KDC which then will look at the part before the realm with the `\@` and will try to figure out from which trusted realm the principal might be coming from and if it found wound it will tell the client to forward the request to this realm. But all of this is special for enterprise principals. But here the proper canonical principal for a user from a trusted domain is needed and hence the realm must match.

bye,
Sumit